### PR TITLE
copied over improved ebuilds due to file rename chaos

### PR DESCRIPTION
**Pull request by mike-kfed**
Here are the newest ebuilds that work. Update for newest version of knxd needs work on Makefile first (doesn't build due to debian/changelog target)

### CODE REVIEW

**Comment by mike-kfed** [comment]
Travis CI fails with the same reason a 0.14.34 ebuild doesn't work `make: *** No rule to make target `debian/changelog', needed by `Makefile.in'.  Stop.` - so that is not my pull-requests fault but something broken in upstream.

**Comment by tru7** [comment]
This is because #463 still applies. I hope @smurfix works on that. I think easiest would be to put changelog into topdir of master branch and adjust Makefile.in and tools/version.sh

**Comment by smurfix** [comment]
I removed the check from the Makefile (for now) and changed version.sh to simply grab the git tag.

**Comment by mike-kfed** [comment]
I added an ebuild 0.14.39 now, since it builds and runs fine on my gentoo box.

**Comment by smurfix** [comment]
Umm, the current `gentoo` branch isn't uptodate with `master`. Please actually merge 0.14.39.

**Comment by mike-kfed** [comment]
how is that important when I add new files and there is no conflict? the ebuild completely separately clones the repo and does everything, it is just a recipe not connected to the other files of knxd in any way. please merge this request

**Comment by smurfix** [comment]
If it's just a recipe that downloads a separate copy, then it probably should be in a branch that doesn't contain the knxd sources in the first place. People shouldn't need to download everything twice.

**Comment by smurfix** [comment]
Anyway, the intent for the local packaging branch is "package the sources in THIS directory so that the distro can install them"; they're here so people can package their local copies, maybe with local modifications.

Can't the ebuild file take the local sources? That's how the Debian and RPM packaging support works …

**Comment by mike-kfed** [comment]
I didn't know the intention of the gentoo branch, you just asked me to make a new merge request for it, so I did. You can delete everything except the gentoo directory, or I can make a new repo only for the ebuilds and we hyperlink to it from the readme, for the people that are interested, what do you think?

ebuilds do not work like .rpm or .deb package building, it's more like BSD ports where it's a recipe how to build the package locally with custom options offered (e.g. I can build knxd without usb support by having set `USE="-usb"`), meaning it has to download the source extra and build extra per machine because the options chosen by the linux-user may be individual = binary packages, if built normally it's just installed, are individual per user.

**Comment by smurfix** [comment]
Well, in my naïve world view (as modified by skimming the Gentoo documentation) an ebuild file is interpreted by bash, thus might it be possible to use something like …
```
src_unpack() {
    git clone "$(pwd)" "$WORKDIR/knxd"
}
```
or something along these lines?

NB, IUSE contains `eibd`. Does that actually make sense? also it's been renamed to "knxd" for about a decade …

**Comment by mike-kfed** [comment]
I can change the IUSE sure.

And these ebuilds already clone from git actually (they are not interpreted by bash, it is python libraries e.g. one for handling git based packages).

**Comment by mike-kfed** [comment]
leaves just one question, gentoo branch here or separate repo for ebuilds?

**Comment by smurfix** [comment]
Really, those shell functions in the ebuild are interpreted by Python?!?

Anyway, I'd leave it here. I'd also add a comment to the README that if you want to build the local sources instead of the archive's, you need to add a `src_unpack` function to the ebuild file that copies your local source to ebuild's workdir.

**Comment by mike-kfed** [comment]
this is not how ebuilds work, they will never use the "local sources" to build, as the portage tree only holds the ebuilds, config files, checksums of the source tarballs and potential patches. You can change it to download the tagged tgz of github, but then it wont build since the version.sh expects to be in a git folder. This would need patching (which can be done in the ebuild, but since the git clone approach works why make it more complicated) so I stayed with git clone.

long story short, they are independent of the source, and intended to be like that. For the python question, I guess they start a shell from python for the shell commands, but in these ebuilds are commands like `emake`, `newinitd`,.... those are not shell commands but implemented in python modules of portage.

**Comment by kripton** [comment]
My 2 cents: Why should there be a separate branch for the ebuilds? Just have them in the main-branch and also have all other recipes/metadata for other distributions there as well. Of course in their own directory structure so they don't clutter the root directory of the repo. This is how other projects handle this as well. I don't think that people would be disturbed by having that stuff in the main branch.

Regarding bash/python and how cloning works: Yes, ebuilds are more or less bash scripts but there are things called "eclasses" that are basically configurable macros. Those ebuilds here use the `git-r3` eclass which makes portage (Gentoo's package manager written in Python) automatically clone the repo. No need to write the cloning instructions in the ebuild again.

Having said that, I currently have knxd running perfectly fine on my Gentoo-based Raspberry Pi. Manually compiled from source ;) The ebuilds in here are using EAPI 5 which is no longer supported (EAPI 8 being the latest). I might have a look at the ebuilds some later time and update them. Thanks for all your work having them here in the first place!

**Comment by smurfix** [comment]
I'll drop the ebuild branch.
